src/java.desktop/share/classes/javax/swing/package-info.java

```diff
@@ -132,15 +132,12 @@
  * For overviews, tutorials, examples, guides, and other documentation,
  * please see:
  * <ul>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/articles-jsp-139072.html"
- *     target="_top">The Swing Connection</a></li>
+ *     <li><a href="https://www.oracle.com/java/technologies/a-swing-architecture.html"
+ *     target="_top">A Swing Architecture Overview</a></li>
  *     <li><a href="https://docs.oracle.com/javase/tutorial/"
  *     target="_top">The Java Tutorial</a></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/training/index.html"
- *     target="_top">Online Training</a>
- *     at the Java Developer Connection <sup>SM</sup></li>
- *     <li><a href="http://www.oracle.com/technetwork/java/javase/tech/index-jsp-142216.html"
- *     target="_top">Java Foundation Classes (JFC)</a> home page</li>
+ *     <li><a href="https://www.oracle.com/java/technologies/javase/training-support.html"
+ *     target="_top">Java SE Training and Certification</a>
```

Comment:
```suggestion
 *     target="_top">Java SE Training and Certification</a></li>
```
The closing `</li>` is not required but it's used in other `<li>` elements in this list.

Reply:
done